pkg/volume/gce_pd/gce_pd_block_test.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce_pd
+
+import (
+	"os"
+	"testing"
+
+	utiltesting "k8s.io/client-go/util/testing"
+)
+
+func TestGetVolumeSpecFromGlobalMapPath(t *testing.T) {
+	// make our test path for fake GlobalMapPath
+	// /tmp symbolized our pluginDir
+	// /tmp/volumeDevices/pdVol1
+	tmpVDir, err := utiltesting.MkTmpdir("volumeDevices")
+	if err != nil {
+		t.Fatalf("can't make a temp dir: %v", err)
+	}
+	//deferred clean up
+	defer os.RemoveAll(tmpVDir)
+
+	path1a := tmpVDir + "/pdVol1"
+	err1a := os.MkdirAll(path1a, 0777)
+	if err1a != nil {
+		t.Fatalf("can't make a temp dir: %v", err1a)
+	}
+	spec, err := getVolumeSpecFromGlobalMapPath(path1a)
```

Comment:
Do you want to validate the fields in this spec?

Reply:
Is that necessary since that is already done within the api/validation and the existing unit tests there?

Reply:
I mean, validate that this function return an appropriate spec with the correct pdName based on the given path.

Reply:
of course, my bad